### PR TITLE
feat: implementing data-providers api

### PR DESCRIPTION
**Pull request by Baoufa**
- adding "/data-providers" route returning Github and POAP providers schemas for the factory

### CODE REVIEW

File: group-generators/helpers/data-providers/github/factory-schema.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          "name": "repository",
+          "type": "string",
+          "example": "sismo-core/sismo-hub",
+          "description": "A unique repository"
+        }
+      ]
+    },
+    {
+      "name": "Organization Members",
+      "functionName": "getOrganizationMembers",
```
Comment (leosayous21) [inline group-generators/helpers/data-providers/github/factory-schema.json:47]: also private

File: group-generators/helpers/data-providers/github/factory-schema.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          "name": "owner",
+          "type": "string",
+          "example": "sismo-core",
+          "description": "A unique organization name"
+        }
+      ]
+    },
+    {
+      "name": "Repository Stargazers",
+      "functionName": "getRepositoryStargazers",
```
Comment (leosayous21) [inline group-generators/helpers/data-providers/github/factory-schema.json:60]: also private

File: group-generators/helpers/data-providers/github/factory-schema.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          "name": "repository",
+          "type": "string",
+          "example": "sismo-core/sismo-hub",
+          "description": "A unique repository"
+        }
+      ]
+    },
+    {
+      "name": "Users with a url",
+      "functionName": "fetchGithubUsersWithUrl",
```
Comment (leosayous21) [inline group-generators/helpers/data-providers/github/factory-schema.json:73]: also private

File: group-generators/helpers/data-providers/poap/factory-schema.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  "providerClassName": "POAPProvider",
+  "functions": [
+    {
+      "name": "Attenders of specific events",
+      "functionName": "queryEventsTokenOwners",
+      "description": "Return the list of Ethereum accounts that attended a list of POAP events",
+      "args": [
+        {
+          "name": "event Ids",
+          "type": "(number | string)[]",
```
Comment (leosayous21) [inline group-generators/helpers/data-providers/poap/factory-schema.json:13]: "(number | string)[]" does not exists in the design system, let use string[]

File: group-generators/helpers/data-providers/poap/factory-schema.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          "example": "63400, 57318",
+          "description": "List of POAP events Ids. https://poap.gallery/"
+        }
+      ]
+    },
+    {
+      "name": "Attenders of a specific event",
```
Comment (leosayous21) [inline group-generators/helpers/data-providers/poap/factory-schema.json:20]: Remove for now as we only use multiple input

File: group-generators/helpers/data-providers/github/factory-schema.json [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+          "name": "repositories",
+          "type": "string[]",
+          "example": "sismo-core/sismo-hub, sismo-core/sismo-protocol",
+          "description": "List of repositories, comma separated"
+        }
+      ]
+    },
+    {
+      "name": "Repository Committers",
+      "functionName": "getRepositoryCommitters",
```
[the remark this answers is not in the document]
Comment (Baoufa) [reply group-generators/helpers/data-providers/github/factory-schema.json]: done

File: group-generators/helpers/data-providers/github/factory-schema.json (hunk 1 above)
Comment (Baoufa) [reply group-generators/helpers/data-providers/github/factory-schema.json]: done

File: group-generators/helpers/data-providers/github/factory-schema.json (hunk 2 above)
Comment (Baoufa) [reply group-generators/helpers/data-providers/github/factory-schema.json]: done

File: group-generators/helpers/data-providers/github/factory-schema.json (hunk 3 above)
Comment (Baoufa) [reply group-generators/helpers/data-providers/github/factory-schema.json]: done

File: group-generators/helpers/data-providers/poap/factory-schema.json (hunk 4 above)
Comment (Baoufa) [reply group-generators/helpers/data-providers/poap/factory-schema.json]: done

File: group-generators/helpers/data-providers/poap/factory-schema.json (hunk 5 above)
Comment (Baoufa) [reply group-generators/helpers/data-providers/poap/factory-schema.json]: done

File: group-generators/helpers/data-providers/index.ts [hunk 7]
```diff
@@ -2,11 +2,13 @@ import BigQueryProvider from "./big-query/big-query";
 import { EnsProvider } from "./ens";
 import { EthLearderboardProvider } from "./eth-leaderboard";
 import { GithubProvider } from "./github";
+import githubFactorySchema from "./github/factory-schema.json";
```
[the remark this answers is not in the document]
Comment (Baoufa) [reply group-generators/helpers/data-providers/index.ts]: done

File: group-generators/helpers/data-providers/index.ts [hunk 8]
```diff
@@ -25,11 +27,16 @@ export const dataProviders = {
...
+  PoapSubgraphProvider,
   RESTProvider,
   SismoSubgraphProvider,
   SubgraphHostedServiceProvider,
   SubgraphDecentralizedServiceProvider,
   SnapshotProvider,
   TransposeProvider,
 };
+
+export const dataProvidersFactorySchemas = [
```
[the remark this answers is not in the document]
Comment (Baoufa) [reply group-generators/helpers/data-providers/index.ts]: done

**Review by leosayous21 — APPROVED** [review]

**Pull request merged**